rskj-core/src/test/java/co/rsk/peg/bitcoin/PocSighashTest.java

```diff
@@ -473,6 +473,67 @@ void test_each_input_sighash_is_unique_for_a_signed_erp_tx_testnet() {
+        FedUtxo utxo2 = FedUtxo.of(new BtcTransaction(networkParameters, Hex.decode(RAW_FUND_TX2)), 0);
+
+        String RAW_FUND_TX3 = "020000000001016a040fa1ec01bac865d4802083a435fef32052091d6d7bd02cc784d00d75e8400100000000fdffffff02916408000000000017a914eaf58ece160a383630667cfc1ccff519ab07c47287beefd305000000002200201ca17d3f37320ac4469faa1119824d98bbec3b1a586174901834d2b842715e7c0400473044022044033baafc0c15c8e2c3128d01f29a7c7c5c6ec790ead8c5be2610efb0e9d581022014b605cb59a5254a189a9fcd35ec94e6f4faae18f7011a41ac7663df33fbfdc50147304402206fc734a5db737ecb37056491881da15c2f8f2ca2bb9e47db7396029b9a0ad5e102206cae6f8766679c666cfa256e886e2816882623167e92fd9e272fd9712aa49d9a0147522102b9bb9ea5c5c0a56421bc7720e81e7f3e5fb9dfcbca946f01d868940efaa5da18210309eb318fe76a6b139d70e4037dd87733ced9e3ed6f851301a88938c7033d473b52ae05272500";
+        FedUtxo utxo3 = FedUtxo.of(new BtcTransaction(networkParameters, Hex.decode(RAW_FUND_TX3)), 0);
+
+        String RAW_FUND_TX4 = "0200000001a223ad7e9d4f5067a8fdddf9e4137af49d5a6dad7ea952533f701061fefd010f020000006a473044022003d999e14be1e2ea15cb16df81e14b30f26c684bcbf6fcbd07975d8648e5f04402207e99a414a0ebf1c3617944610de5ea002439a1ac56e667a60c095184c2419559012102c6bf1e099ec95510a8da3d1c67026cb65a019bad57dc2255dd56a426c629327bfdffffff0300000000000000001b6a1952534b540162db6c4b118d7259c23692b162829e6bd5e4d5b0891300000000000017a914eaf58ece160a383630667cfc1ccff519ab07c4728791200000000000001976a9145952b24450e80668e069b8152a3a38ea7f6ad44c88ace7262500";
+        FedUtxo utxo4 = FedUtxo.of(new BtcTransaction(networkParameters, Hex.decode(RAW_FUND_TX4)), 1);
+
+        List<FedUtxo> utxos = Arrays.asList(utxo1, utxo2, utxo3, utxo4);
+        Coin fees = Coin.valueOf(10_000);
+        Coin totalUtxosAmount = utxos.stream().map(fedUtxo -> fedUtxo.btcTransaction.getOutput(fedUtxo.getOutputIdx()).getValue())
+            .reduce(
+                Coin.ZERO, Coin::add
+            );
+        Coin pegoutTxAmount = totalUtxosAmount.minus(fees);
+
+        Address destinationAddress = Address.fromBase58(networkParameters, "2MtYSUzFWEQV62r92bsGX8ewE5Mgpv4xn9M");
+
+        BtcTransaction pegoutTx = new BtcTransaction(networkParameters);
+        pegoutTx.addOutput(pegoutTxAmount, destinationAddress);
+        pegoutTx.setVersion(2);
+
+        pegoutTx.getInputs().forEach((transactionInput -> {
+            Script redeemScriptFromInput = getRedeemScriptFromInput(transactionInput);
+            assertEquals(fed.getRedeemScript(), redeemScriptFromInput);
+        }));
+    }
```

Comment:
Do We Also Need To Test The New Method Here: **`getFirstInputSigHash`** ?

Reply:
Tests were refactored and the way I assert the sighash is valid was improved. Please check it out.